tests/package/validate/test_general.py

```diff
@@ -345,3 +345,33 @@ def test_validate_package_with_resource_data_is_a_string_issue_977():
     assert report.flatten() == [
         [None, None, None, "package-error"],
     ]
+
+
+def test_validate_package_with_missing_values_993():
+    package = Package(descriptor="data/package-with-missingvalues-993.json")
+    output = list(
+        map(
+            lambda metadata_error: [metadata_error.code, metadata_error.note],
+            package.metadata_errors,
+        )
+    )
+    assert output == [
+        [
+            "resource-error",
+            '"missingValues" should be set as "resource.schema.missingValues" (not "package.missingValues").',
+        ]
+    ]
+
+
+def test_validate_package_with_fields_993():
+    package = Package(descriptor="data/package-with-fields-993.json")
+    output = list(
```

Comment:
Can we just write assertions like this:
```
assert package.metadata_errors[0].code == ...
assert package.metadata_errors[0].note == ...
```
Your version is smart but really makes the reader think while the simpler one doesn't require any additional energy

Reply:
Thank you for pointing out. I was making it complex unnecessarily :).